spark/src/main/scala/org/apache/spark/sql/delta/files/TransactionalWrite.scala

```diff
@@ -348,6 +354,7 @@ trait TransactionalWrite extends DeltaLogging { self: OptimisticTransactionImpl
   def writeFiles(
       inputData: Dataset[_],
       writeOptions: Option[DeltaOptions],
+      isOptimize: Boolean,
```

Comment:
can you clarify why we need this new isOptimize flag now ?

Reply:
Added some comments.